### ISSUE TITLE
The number of available endpoints needs to be parameterized in the USB DWC driver

### ISSUE THREAD

**Issue by stoyan-shopov** [libopencm3/libopencm3#1260]
Currently, the USB DWC driver has a hard coded number of handled endpoints, 4. This is not always true, e.g., for an STM32F723 device.

https://github.com/libopencm3/libopencm3/blob/08c5f2a1fb16b279907eb622ac043046d7a1a35c/lib/usb/usb_dwc_common.c#L122

https://github.com/libopencm3/libopencm3/blob/08c5f2a1fb16b279907eb622ac043046d7a1a35c/lib/usb/usb_dwc_common.c#L342

What should be the proper way to parameterize the endpoint count?

**Comment by karlp** [comment]
Correct, I think there's even an old ticket on this, but I'll leave this open for now.
